From 9b1a7cde36c8f9f4159ad4fdf3f87d8d84dc2f6e Mon Sep 17 00:00:00 2001
From: Kirubagharan A S <kirubagharanas@ami.com>
Date: Sun, 15 Mar 2026 19:15:00 +0000
Subject: [PATCH] service: use boost::asio::io_context

Newer Boost ASIO uses io_context instead of io_service.
Switch to io_context to fix compilation with current toolchains.

Signed-off-by: Kirubagharan A S <kirubagharanas@ami.com>
---
 service/src/mainapp.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/service/src/mainapp.cpp b/service/src/mainapp.cpp
index 4f8fb7c..d6696a8 100644
--- a/service/src/mainapp.cpp
+++ b/service/src/mainapp.cpp
@@ -676,7 +676,7 @@ void checkPFRandAddObjects(sdbusplus::asio::object_server& server,
 int main()
 {
     // setup connection to dbus
-    boost::asio::io_service io;
+    boost::asio::io_context io;
     auto conn = std::make_shared<sdbusplus::asio::connection>(io);
     pfr::stateTimer = std::make_unique<boost::asio::steady_timer>(io);
     pfr::initTimer = std::make_unique<boost::asio::steady_timer>(io);
-- 
2.34.1
